lib/app/endpoint/home.rb

```diff
@@ -164,46 +171,45 @@ class Home < Endpoint
           end
         end
 
-        # Handles EHR Launch and any redirects
         get '/:id/:key/:endpoint/?' do
           instance = Inferno::Models::TestingInstance.get(params[:id])
-          halt 404 unless !instance.nil? && instance.client_endpoint_key == params[:key] && %w[launch redirect].include?(params[:endpoint])
+          halt 404 unless !instance.nil? && instance.client_endpoint_key == params[:key] && ['launch','redirect'].include?(params[:endpoint])
 
           sequence_result = instance.waiting_on_sequence
 
           if sequence_result.nil? || sequence_result.result != 'wait'
-            redirect "/#{base_path}/#{params[:id]}/?error=no_#{params[:endpoint]}"
+            redirect "/#{BASE_PATH}/#{params[:id]}/?error=no_#{params[:endpoint]}"
           else
-            klass = Inferno::Sequence::SequenceBase.subclasses.find do |x|
-              x.name.demodulize.start_with?(sequence_result.name)
-            end
+            klass = instance.module.sequences.find{|x| x.name.demodulize == sequence_result.name}
```

Comment:
Use the `sequence_name` method you created
https://github.com/siteadmin/inferno/blob/ccef0fbef3856ad1f1a1787f3e9bb6288cdbad54/lib/app/sequence_base.rb#L204-L206
```suggestion
            klass = instance.module.sequences.find{|x| x.sequence_name == sequence_result.name}
```